enhancements/node-tuning/mixed-cpu-node-plugin.md

```diff
@@ -0,0 +1,352 @@
+This means that when the feature enabled, Kubelet's `reservedSystemCpus` composed of PerformanceProfile's 
+`spec.cpu.reserved` + `spec.cpu.shared`.
+The reason we utilize Kubelet `reservedSystemCpus` is because of CPU manager, which is not aware of the CPUs
+lying in this pool, so it doesn't undo or changes the allocation logic performed by the plugin.
+Therefore, the plugin assigns the shared CPUs to containers without racing/conflict
+the CPU manager behavior. 
+More about this decision at the [Alternative](mixed-cpu-node-plugin.md#alternatives) section
+
+There is a plan for a feature called [shared-partition](https://github.com/openshift/enhancements/pull/1421) 
+that implements additional pool.
+When shared-partition lands, Kubelet's cpu pool layout should look roughly like:
+1. `reservedSystemCpus`
+2. `guaranteedCpus` <- new pool
+3. `sharedCpus` = All CPUs - `reservedSystemCpus` - `guaranteedCpus`
+
+When shared-partition feature enabled, 
+mixed-cpu-node-plugin allocates the shared cpus from the `sharedCpus` pool and not
+from the reservedSystemCpus anymore.
+
+The following section summarizes the shared-cpus source in each flow:
+* mixed-cpu enabled -> `cpu.shared` appends to Kubelet's `reservedSystemCPUs`
+* shared-partition enabled -> Kubelet's `sharedCpus` = All cpus - `reserved` - `isolated`
+* Both enabled -> sharedCpus = All cpus - `reserved` - `isolated`
+
+The node-plugin populates a special device named `openshift.io/shared-cpus` to provide way for pods to request
+for this special type of CPUs.
+There's no meaning to the value/number of `openshift.io/shared-cpus` devices that the pod requests.
+For example:
+```yaml
+requests:
+   openshift.io/shared-cpus: 1
+```
+Or
+```yaml
+requests:
+   openshift.io/shared-cpus: 2
+```
+Are the same in that context, the container has access to the same number of shared-cpus in both cases.
+In other words, the device request only uses as a hint to tell the plugin that shared-cpus required for the container and
+does not indicate any actual value.
+
+The reason for specifying a device, and not, for example, an annotation,
+is because when application pod requests for a device, the scheduler makes the pod pending till device up and running.
+This gives the node plugin room for setup and not being depended on pod admission order.
+So only when the node plugin finishes the registration process with NRI and device plugin, the scheduler
+admits the application pod.
+
+Once a container under the pod requests and gets access to shared cpus, new environment variable named
+`OPENSHIFT_SHARED_CPUS=<CPU-IDs>` will be present under the container's environment variables.
+
+The mixed-cpu-node-plugin itself is essentially a pod that runs as a DaemonSet on the worker nodes.
+cluster-node-tuning-operator (NTO) will be modified to support the deployment of the DaemonSet and additional
+resources (ServiceAccount, RBAC resources, SecurityContextConstraint, etc.)
```

Comment:
what role will this service account hold?  does this daemon need to report status to a component?  if so, how and on what resource?

Reply:
> what role will this service account hold?

The plugin needs to be running at `Privileged`  mode in order to communicate with the runtime, so the service account hold the role to use the SCC that allows `Privileged` security-context:
https://github.com/openshift-kni/mixed-cpu-node-plugin/blob/main/pkg/manifests/yamls/role.yaml 

> does this daemon need to report status to a component? if so, how and on what resource?
Yes. it should report its status to NTO's and the status should be listed as part of the `performanceProfile` status